pilot/pkg/xds/sds.go

```diff
@@ -51,8 +52,10 @@ type SecretResource struct {
 
 var _ model.XdsCacheEntry = SecretResource{}
 
-func (sr SecretResource) Key() string {
-	return sr.SecretResource.Key() + "/" + sr.pkpConfHash
+func (sr SecretResource) Key() model.KeyHash {
+	h := hash.New()
+	h.Write([]byte(sr.SecretResource.Key() + "/" + sr.pkpConfHash))
```

Comment:
I had initially made ethe same PR but this gave me some pause. If there is a collision here it's a CVE (note we literally had this CVE before, just jkt from hashing - we. missed namespace in the key)

Do we have a 100% guarantee there is no collision? I am assuming no which makes this a bit worrisome and can leak secrets across namespace.

Reply:
In theory this could push more towards having a cache per type, which could reduce lock contention and allow the Peer authentication stuff to be trivial (just clear the whole eds cache when PA changes)

Reply:
I am totally ok to isolate different caches. they could use the existing xds cache impl.